DCOS-42490 - Pull Request Template maintenance.

orsenthil opened the pull request:
## High-level description

* DCOS-42490 - Pull Request Template maintenance.
* Address review comments from https://github.com/dcos/dcos/pull/5541 
* Address Formatting issues that were visible after the previous PR was merged.

orsenthil commented:
@mesosphere-mergebot label  ready for review

pcrews commented on PULL_REQUEST_TEMPLATE.md:
```diff
@@ -55,14 +55,13 @@ If you are changing components or packages in DC/OS (e.g. you are bumping the sh
 Reviewers should be:
 
 * Developers who understand the code being modified.
-* Developers responsible for code that interacts with or depends on the code being modified.
```
I think the original makes more sense:
Reviewers should be: "Developers responsible for <any> code that interacts..." NOTE: <any> is a suggested addition
vs.:
Reviewers should be "Developers are responsible for the code..."

orsenthil replied:
@pcrews  - Yeah, I agree. I will revert this statement back.

jgehrcke commented on PULL_REQUEST_TEMPLATE.md:
```diff
@@ -10,12 +14,10 @@ What features does this change enable? What bugs does this change fix?
-a comment will be added, and the FixVersion field of the JIRA will be set to DC/OS release version upon merged.
-The FixVersion will correspond to the [fix_version_map](https://github.com/dcos/dcos/blob/master/mergebot-config.json#L30) 
-of the mergebot-config.
+Please provide a list of JIRA tickets that this Pull Request acts upon. This section will be parsed, a comment will be added, and the FixVersion field of the JIRA will be set to DC/OS release version upon merged.
```
This is great. I have tried to re-order this a little bit to make it hopefully even easier to read! Proposal:

"""
This section will be parsed for JIRA tickets. Please provide a list of JIRA tickets that this Pull Request is supposed to act upon. Upon merge, every JIRA ticket listed here will be modified by Mergebot:
- a comment will be added
- the `fixVersion` field of the JIRA ticket will be set to the currently configured DC/OS release version.
"""

orsenthil replied:
I have adopted this with a slight tweak. I have added this to the header section of the PR and presented the current parsing behavior so that it is clear to all the users and is presented upfront. Thank you!

jgehrcke commented:
Good stuff! If still possible let's maybe consider the comments I just added to https://github.com/dcos/dcos/pull/5541.

orsenthil commented:
Hi @jgehrcke, @pcrews  - I have addressed the comments here, and also the ones mentioned in #5541 
I have rolled up all the comments into the header section so that it is easier to read and explained the current parsing behavior.

PTAL. Thank you!
